osgar/drivers/kloubak.py

```diff
@@ -410,13 +411,27 @@ def update_pose(self):
         diff = [e - prev for e, prev in zip(self.encoders, self.last_pose_encoders)]
         self.last_pose_encoders = self.encoders
         if self.num_axis == 2:
+            if self.last_join_angle is None:
+                return False
+            if self.prev_joins_angle is None:
+                self.prev_joins_angle = self.last_join_angle
+                return False
+            join_angle_diff = self.last_join_angle - self.prev_joins_angle
```

Comment:
I still do not understand the purpose of this `join_angle_diff`

Reply:
The core of this change is that i require a heading of the front axle. The change of the heading is then given by the rear heading change and change of the joint(s) angle (`join_angle_diff`).